lib/html-encoding-sniffer.js

```diff
@@ -37,11 +37,10 @@ function prescanMetaCharset(buffer) {
         i += 4;
```

Comment:
>  (The two 0x2D bytes can be the same as those in the '<!--' sequence.)

Reply:
Oh, we're backtracking, so this should work. Nvm me.

Reply:
Yes, thus the minus-1/minus-2 in the next few lines.